book/src/development/README.md

```diff
@@ -0,0 +1 @@
+# Clippy Development
```

Comment:
I think we can add some text here from the CONTRIBUTING.md, but I'm not sure what exactly, yet. Same for Infrastructure.